test_pull_requests

```diff
@@ -152,6 +178,97 @@ def submitted_tests_for_branch(branch)
   $submitted_tests[branch] ? $submitted_tests[branch] : $submitted_tests['*']
 end
 
+# Pulls out as much metadata from the bot comment as it can, including
+# state (AKA content), Jenkins build URL, base repo commit ID, and the
+# extended tests metadata
+def extract_bot_comment_fields(bot_comment_body, settings=nil)
+  test_prefix = nil
+  content = nil
+  build_url = nil
+  base_commit = nil
+  extended_tests = nil
+  state = nil
+  build_url_pattern = nil
+  if settings.nil?
```

Comment:
I'd break out this `if` block into a function or two, one for the build pattern, one for the prefix. Readability over the "performance" loss in having to re-eval the `if` a couple times.

ex:
```rb
def determine_test_prefix(bot_comment_body, settings=nil)
  return settings['test_prefix'] unless settings.nil?
  
  test_prefix = nil
  if bot_comment_body =~ /^([^:(]+): /
    test_prefix = $1.strip
  elsif bot_comment_body =~ /^([^ (]+) /
    test_prefix = $1.strip
  end
  test_prefix
end
```